numba/tests/test_comprehension.py

```diff
@@ -251,6 +225,166 @@ def list24(x):
             msg = "Invalid usage of == with parameters"
             self.assertIn(msg, str(raises.exception))
 
+class TestArrayComprehension(unittest.TestCase):
+
+    @tag('important')
+    def test_comp_with_array_1(self):
```

Comment:
The following code works:
```
a = np.empty(10, dtype=np.complex64)
a[0] = 1.j
```
However, the problem is how do we produce an IR for `empty(..., dtype=...)` before type inference? The information about element being `complex64` is only known after inference.

Similar situation is with list comprehension, where `build_list` comes before `list.append(...)`. However, for list this works out with no problem (before array-comprehension, or set `enable_inline_arraycall = False`):

```
l = [ 1j * x for x in range(10) ]
a = np.array(l)
```

So one way to solve this would require some tweaking in type inference to fill in the correct `dtype` data if it is missing from the original `array(..)` call. But this, however, is not quite the semantics of numpy. Any thoughts? @sklam